collection formatter generalized to "OxfordStyle" and "RegularStyle"

ThomasMentzel opened the pull request:
This change is regarding the concatenation of multiple items with comma and e.g. "and". See:

new []{"Peter", "Paul", "Mary"} => EN => "Peter, Paul, and Mary"  (if you use Oxford style)
new []{"Peter", "Paul", "Mary"} => EN => "Peter, Paul and Mary"   (if you dont use Oxford style)
new []{"Peter", "Paul", "Mary"} => DE => "Peter, Paul and Mary"   (this is how it works in italian and german)

The English and Italian version got generaltized to "OxfordSyle" and "RegularStyle". The registry now uses these base classes with a constructor parameter to get the default separator.
The registry now registers a german collection formatter.

hazzik commented:
I'm :-1: on this because "Oxford Style" is too localalised for Humanizer.

hazzik commented:
Actually this is a good change after close look. Sorry about that

mexx commented:
@ThomasMentzel please add this PR to release notes according to [CONTRIBUTING.md](https://github.com/MehdiK/Humanizer/blob/master/CONTRIBUTING.md) with message like "added German collection formatter".

ThomasMentzel commented:
I added the required information. I added additional languages according to [wikipedia] (https://en.wikipedia.org/wiki/Serial_comma#Other_languages) and google translator .

MehdiK commented:
Very nice addition. Thanks. This is now rebased and pushed.

MehdiK commented:
This is now released to NuGet as v1.34.0. Thanks.

Bartmax commented:
DefaultDeparator ??

MehdiK commented:
Sorry @Bartmax - not sure what you mean! Can you please elaborate a bit?
